pinot-common/src/main/java/org/apache/pinot/common/minion/MinionTaskMetadataUtils.java

```diff
@@ -79,10 +80,38 @@ public static void deleteTaskMetadata(HelixPropertyStore<ZNRecord> propertyStore
     }
   }
 
+  /**
+   * Deletes the minion task metadata ZNRecord for the given tableName, from both the new path
+   * MINION_TASK_METADATA/${tableNameWthType} and the old path
+   * MINION_TASK_METADATA/<any task type>/${tableNameWthType}
+   */
+  public static void deleteTaskMetadata(HelixPropertyStore<ZNRecord> propertyStore, String tableNameWithType) {
+    // delete the minion task metadata ZNRecord MINION_TASK_METADATA/${tableNameWthType}
+    String path = ZKMetadataProvider.constructPropertyStorePathForMinionTaskMetadata(tableNameWithType);
+    if (!propertyStore.remove(path, AccessOption.PERSISTENT)) {
+      throw new ZkException("Failed to delete task metadata for table: " + tableNameWithType);
+    }
+    // delete the minion task metadata ZNRecord MINION_TASK_METADATA/<any task type>/${tableNameWthType}
+    List<String> childNames =
+        propertyStore.getChildNames(ZKMetadataProvider.getPropertyStorePathForMinionTaskMetadataPrefix(),
+            AccessOption.PERSISTENT);
+    if (childNames != null && !childNames.isEmpty()) {
+      for (String child : childNames) {
+        // Even though some child names are not task types (e.g., in the new metadata path, the child name
+        // is a table name), it does not harm to try to delete the non-existent constructed path.
+        String oldPath =
+            ZKMetadataProvider.constructPropertyStorePathForMinionTaskMetadataDeprecated(child, tableNameWithType);
+        if (!propertyStore.remove(oldPath, AccessOption.PERSISTENT)) {
```

Comment:
Hmmm, I tested locally using integration test, if we are trying to remove a non-existent path, it returns true. 

Could you please let me know your test setup so I can try it myself? Thanks!

Reply:
I have another version implemented locally 😄 , which scans jars and find all minion task types, similar to what https://github.com/apache/pinot/blob/master/pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/TaskGeneratorRegistry.java does.

I chose the current one (which reads from ZK) because I want to use ZK as the source of truth, so we will not miss any ZNode. 

But you point out a very good point that there might be thousands of tables. I can add a TODO and we can decide later.

Reply:
Let me know your thought, we can switch to the other approach or we can merge the current approach and get back to it later.

(BTW, could you please merge it if we want to go with the current approach as for now because I cannot merge the code 😄 )

Reply:
We can get back to it later. I'll merge this one. Thanks for making the change!